activerecord/lib/active_record/associations/collection_proxy.rb

```diff
@@ -928,7 +928,20 @@ def include?(record)
         !!@association.include?(record)
       end
 
-      def proxy_association # :nodoc:
+      # Returns the association for the collection.
+      #
+      #   class Person < ActiveRecord::Base
+      #     has_many :pets
+      #   end
+      #
+      #   person.pets.proxy_association
+      #   # => #<ActiveRecord::Associations::HasManyAssociation owner="#<Person:0x00>">
+      #
+      # Returns the same object as <tt>person.association(:pets)</tt>,
```

Comment:
`association` isn't a public method so I don't think we should mention it.
Maybe:
```suggestion
      # Returns the Association object,
```

Reply:
It's documented here:
https://github.com/rails/rails/blob/f6918a5b5d0b8c3a700513c2624cc195e0a78d14/activerecord/lib/active_record/associations.rb#L719-L732

I linked to this section from the method docs, but that is a good point.

> Returns the Association object,
Association doesn't exist and Associations is also private. :thinking:

Reply:
We should make `association` public as well along with owner, reflection and target